### PR TITLE
Feat: Added Home Page

### PR DESCRIPTION
**Pull request by spraveenofficial**
Added Home Page.
Added Protected & Guest Routes.
Added Sidebar & Right SideBar for App.
Added Feed with Infinite Scroll.
Added Like and Dislike Feature.
Added Bookmark and Remove from HomePage Feed.
Added Search User with Debounce.
Added Reducers for all of the above Features.
Added Animated Modal.
Added Logout Functionality from Sidebar.

`--MODAL --`

`--HOMEPAGE --`

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,13 +1,43 @@
...
       <Routes>
-        <Route path="login" element={<Login />} />
-        <Route path="signup" element={<Signup />} />
+        <Route element={<ProtectedRoutes />}>
+          <Route path="/" element={<HomeComponent />} />
+        </Route>
+        <Route element={<GuestRoutes />}>
+          <Route path="signup" element={<Signup />} />
+          <Route path="login" element={<Login />} />
+        </Route>
```
Comment (shivanigangadharan) [inline src/App.js:40]: Can import all the routes from an AppRoutes.js file to make code more readable here.

File: src/Components/CreatePost/index.jsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            cols="50"
+            placeholder="What's happening?"
+            onChange={(e) => setCaption(e.target.value)}
+            value={caption}
+          ></textarea>
+        </div>
+      </div>
+      {selectedImage !== null && (
+        <div className="relative">
+          <img src={selectedImage} alt="" className="w-full h-80 pr-4 pl-4" />
```
Comment (shivanigangadharan) [inline src/Components/CreatePost/index.jsx:112]: add alt message here

File: src/Components/Shared/Promotion.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+              strokeWidth="2"
+              d="M17 8l4 4m0 0l-4 4m4-4H3"
+            />
+          </svg>
+        </button>
+      </div>
+      <div className="col-span-2 relative">
+        <img
+          src="https://www.digitalocean.com/_next/static/media/cloudJourneyImage.954519ea.svg"
+          className="absolute bottom-0 right-0"
```
Comment (shivanigangadharan) [inline src/Components/Shared/Promotion.jsx:37]: `alt` message here too

**Review by cjinc121 — APPROVED** [review]
good to merge

**Review by shivanigangadharan — APPROVED** [review]
Suggested few minor changes, other than that code looks very good. I like how you've created the constants file separately to make the code clean and easier to understand. Great work 👍🏼

File: src/Components/Shared/Promotion.jsx (hunk 3 above)
Comment (spraveenofficial) [reply src/Components/Shared/Promotion.jsx:37]: Done. Thank you.

File: src/Components/CreatePost/index.jsx (hunk 2 above)
Comment (spraveenofficial) [reply src/Components/CreatePost/index.jsx]: Done.

**Pull request merged**